Program.cs

```diff
@@ -39,20 +49,37 @@ static void Main(string[] args)
                     module.ProcessSequence(sequence);
                 }
 
-                if (++sequencesProcessed % 100000 == 0)
+                On(Settings.ShowProgress, () => progressBar.Update());
+                On(Settings.Verbose, () =>
                 {
-                    Console.Error.WriteLine($"{sequencesProcessed} sequences completed ~{sequenceReader.ApproximateCompletion()}%");
-                }
+                    if (sequenceReader.SequencesRead % UpdatePeriod == 0)
+                    {
+                        Console.Error.WriteLine($"{sequenceReader.SequencesRead.WithSsiUnits()} sequences completed ({sequenceReader.ApproximateCompletion:0.0}%)");
+                    }
+                });
             }
 
-            Console.Error.WriteLine($"{sequencesProcessed} sequences processed");
+            var results = new Dictionary<string, object>()
```

Comment:
oh cute, probably add a match to pull in the source files as additional metadata

Reply:
added